lib/sensu-plugin/utils.rb

```diff
@@ -74,15 +77,35 @@ def map_v2_event_into_v1(orig_event = nil)
           ##
           # Fill in missing client attributes
           ##
-          event['client']['name']        ||= event['entity']['id']
+
           event['client']['subscribers'] ||= event['entity']['subscriptions']
 
+          ##
+          #  Map entity metadata into client attributes
+          #  Note this is potentially destructive as it may overwrite existing client attributes.
+          ##
+          if event['entity'].key?('metadata')
+            ##
+            #  Map metadata label 'name' to client name attribute
+            ##
+            event['client']['name'] ||= event['entity']['metadata']['name']
+
+            ##
+            #  Map special metadata label defined in map_label as json string and convert to client attributes
+            #  Note this is potentially destructive as it may overwrite existing client attributes.
+            ##
+            if event['entity']['metadata'].key?('labels') && event['entity']['metadata']['labels'].key?(map_label)
+              json_hash = JSON.parse(event['entity']['metadata']['labels'][map_label])
+              event['client'].update(json_hash)
+            end
+          end
+
           ##
           # Fill in renamed check attributes expected in 1.4 event
           #   subscribers, source
           ##
           event['check']['subscribers'] ||= event['check']['subscriptions']
-          event['check']['source'] ||= event['check']['proxy_entity_id']
+          event['check']['source'] ||= event['check']['proxy_entity_name']
 
           ##
           # Mimic 1.4 event action based on 2.0 event state
```

Comment:
needs update

Reply:
all references to 2.x and 1.x removed convered to Go and Ruby respectively in latest commit.